## Session 3 Assignment:                                                                              
Formulate both null and alternative hypothesis for each and solve using the techniques used in session 3 (using python).

## Question 1: 

A pharmaceutical company conducts a clinical trial for a new drug and finds that out of 500 patients, 180 show signs of improvement in their condition. Test whether there is a significant difference in the proportion of patients showing improvement compared to the expected response rate of 30%.

#### Here are the null and alternative hypotheses for this problem:
 H0: p = 0.3  (The proportion of patients showing improvement is equal to the expected response rate of 30%).
 Ha: p ≠ 0.3 (The proportion of patients showing improvement is not equal to the expected response rate of 30%).

In [1]:
import numpy as np
from scipy import stats
from scipy.stats import norm
from statsmodels.stats.proportion import proportions_ztest

In [2]:
# Defining variables

n = 500  #Total number of patients
x = 180 # Number of patients showing improvement
p0 = 0.3 #Expected response rate

value = p0 * n # expected person to show improvement

z_score, p_value = proportions_ztest(x,n,p0)

print('30% population: ', value)
print('z-score is: {:.3f}'.format(z_score))
print('P-value is: {:.3f}'.format(p_value))

alpha = 0.05 # for 95% confidence interval

if p_value < alpha:
    print("Reject the null hypothesis")
else: 
    print("Failed to reject the null hypothesis")
    

30% population:  150.0
z-score is: 2.795
P-value is: 0.005
Reject the null hypothesis


### Answer: 

Based on p-value = 0.005, we reject the null hypothesis, Hence, there is significant difference between expected responce rate and patients showed sign of improvement for a new drug. 

## Question 2: 

A pharmaceutical company is developing a new drug claimed to reduce blood pressure by 10 mmHg. The company selects a sample of patients and administers the drug. After treatment, the mean reduction in blood pressure is found to be 9.2 mmHg with a standard deviation of 2.5 mmHg. Conduct a hypothesis test to determine if there is significant evidence to support the claim that the drug reduces blood pressure by 10 mmHg.

Given Data: 9.2, 9.5, 8.8, 10.1, 9.6, 8.9, 9.3, 9.7, 9.1, 9.4

In [3]:
# Given data

BP_data = np.array([9.2, 9.5, 8.8, 10.1, 9.6, 8.9, 9.3, 9.7, 9.1, 9.4])
sample_mean = BP_data.mean()
print(sample_mean)

hypothesized_mean = 10


9.36


In [4]:
population_std_dev = 2.5  # Assume a known population standard deviation
z_statistic = (sample_mean - hypothesized_mean) / (population_std_dev / np.sqrt(len(BP_data)))
p_value_z = norm.cdf(z_statistic) * 2  # Two-tailed test
print('p-value is: {:.3f}'.format(p_value_z))

alpha = 0.05 # for 95% confidence interval

if p_value_z < alpha:
    print("Reject the null hypothesis")
else: 
    print("Failed to reject the null hypothesis")

p-value is: 0.418
Failed to reject the null hypothesis


### Answer:
Based on p-value = 0.418, we failed to reject the null hypothesis, Hence, there is no significant difference between measured reductio in blood pressure and reported by company. 

## Question 3: 

A biotechnology company is investigating the variability of gene expression levels under two different environmental conditions. They measure the expression levels of a particular gene in 20 samples for each condition and compute the sample variances. Test whether there is a significant difference in the variance of gene expression levels between the two conditions.

Given data
Condition 1: 4.5, 5.1, 3.9, 4.8, 4.4, 5.2, 4.7, 4.6, 4.9, 5.0 # Expression levels condition 1
Condition 2: 3.8, 3.7, 4.0, 3.9, 4.1, 3.6, 4.0, 3.8, 4.2, 3.9 # Expression levels condition 2

#### Here are the null and alternative hypotheses for this problem:
 H0: There is no significant difference between the variances of gene expression levels under the two conditions.
 H1: There is a significant difference between the variances of gene expression levels under the two conditions.
 We can test this using an F-test in Python:

In [5]:
# Given Data
condition_1 = [4.5, 5.1, 3.9, 4.8, 4.4, 5.2, 4.7, 4.6, 4.9, 5.0]  
condition_2 = [3.8, 3.7, 4.0, 3.9, 4.1, 3.6, 4.0, 3.8, 4.2, 3.9]

# Compute variances
var_1 = np.var(condition_1)
var_2 = np.var(condition_2)

# Perform F-test
f_stat, p_value_f = stats.f_oneway(condition_1, condition_2)

print('variance of condition 1: {:.6f}'.format (var_1))
print('variance of condition 2: {:.6f}'.format (var_2))
print('p-value is: {:.6f}'.format(p_value_f))

alpha = 0.05 # for 95% confidence interval

if p_value_f < alpha:
    print("Reject the null hypothesis")
else: 
    print("Failed to reject the null hypothesis")

variance of condition 1: 0.132900
variance of condition 2: 0.030000
p-value is: 0.000011
Reject the null hypothesis


### Answer:
Based on p-value = 0.000011, we reject the null hypothesis, Hence, there is significant difference between in variance of condition 1 and condition 2.

## Question 4: 

A life sciences company is comparing two different protocols for cell culture growth rates. One protocol involves a newly formulated growth medium, while the other uses a standard medium. The company takes samples from two groups of cultures and measures the growth rates. Test whether there is a significant difference in the mean growth rates between these two protocols.

Given data
Group 1: 12, 13, 11, 15, 14, 10, 12, 14, 13, 12, 16, 13, 14, 15, 12 # Growth rates group 1
Group 2: 10, 11, 10, 12, 11, 9, 12, 11, 10, 10, 13, 11, 12, 13, 11 # Growth rates group 2

In [6]:
# Data
group_1 = [12, 13, 11, 15, 14, 10, 12, 14, 13, 12, 16, 13, 14, 15, 12]
group_2 = [10, 11, 10, 12, 11, 9, 12, 11, 10, 10, 13, 11, 12, 13, 11] 

# Perform t-test
t_stat, p_value_ttest = stats.ttest_ind(group_1, group_2)

print('T-statistic: {:.3f}'.format(t_stat)) 
print('P-value: {:.3f}'.format(p_value_ttest))

alpha = 0.05 # for 95% confidence interval

if p_value_ttest < alpha:
    print("Reject the null hypothesis")
else: 
    print("Failed to reject the null hypothesis")

T-statistic: 3.878
P-value: 0.001
Reject the null hypothesis


### Answer:

a p-value of 0.001 for this two-sample t-test, Hence, we reject the null hypothesis and new growth medium protocol results in a significantly different mean growth rate compared to the standard protocol.